[jmsandiegoo] iP

jmsandiegoo opened the pull request:
# Tyrone
> "Your mind is for having ideas, not holding them." - David Allen ([source](https://dansilvestre.com/productivity-quotes/))

Tyrone is a chatbot companion that helps free your mind of having to remember things you need to do. It's,

- text-based
- easy to learn
- ~~FAST~~ *SUPER* FAST to use

All you need to do is,

1. download it from [here](https://github.com/jmsandiegoo/ip/releases/tag/A-Jar).
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you 🤓

And it is **FREE!**

Features:

- [x] Managing tasks
- [ ] Managing deadlines (coming soon)
- [ ] Reminders (coming soon)

If you are a Java programmer, you can use it to practice Java too. Here's the `main` method:

```java
public class Tyrone {
    public static void main(String[] args) {
        Application.launch(Tyrone.class, args);
    }
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

Commit:
docs/README.md: Tweak document template

Let's tweak the docs/README.md (which is used as the user guide)
to fit Duke better. Specifically,

1. mention product name in the title
2. mention adding a product screenshot and a product intro
3. tweak the flow to describe feature-by-feature

jmsandiegoo committed:
Renamed Bot to Tyrone and updated personality message

jmsandiegoo committed:
Added echo and bye functionality

jmsandiegoo committed:
changed input reader to scanner

jmsandiegoo committed:
add and display list function done

jmsandiegoo committed:
Added TodoItem to support more functionality for todo item

jmsandiegoo committed:
Updated TodoList and Tyrone classes to support marking and unmarking feature

jmsandiegoo committed:
Fixed some print formatting issues

jmsandiegoo committed:
Fixed unmark cmd freeze bug

jmsandiegoo committed:
rename classes to suit the three types of tasks

jmsandiegoo committed:
Added three subtypes of Tasks

jmsandiegoo committed:
updated tyrone and tasklist to support three task functionality

jmsandiegoo committed:
implemented cli text ui testing

jmsandiegoo committed:
Added command not exist error handling

jmsandiegoo committed:
Implemented enums for commands, implemented overall error handling

jmsandiegoo committed:
Fixed error messages output and typos

jmsandiegoo committed:
Updated test case with error handling

jmsandiegoo committed:
delete feature added

jmsandiegoo committed:
updated test to include delete feature

jmsandiegoo committed:
Update README.md

jmsandiegoo committed:
Update README.md

jmsandiegoo committed:
Tyrone class: update method to private except main

jmsandiegoo committed:
Implement load file without error handling

jmsandiegoo committed:
Add task list local saves with basic error handling

jmsandiegoo committed:
Merge branch 'branch-Level-7'

jmsandiegoo committed:
Merge branch 'master' of github.com:jmsandiegoo/ip

jmsandiegoo committed:
Add datetime support for deadline event

jmsandiegoo committed:
Add datetime support for event task

jmsandiegoo committed:
Update text-ui-test to include new features

jmsandiegoo committed:
Fix event command exception handling

jmsandiegoo committed:
Merge branch 'branch-Level-8'

jmsandiegoo committed:
Update .gitignore to include data dir

jmsandiegoo committed:
Refactor DateTime class

jmsandiegoo committed:
Add and implement UI class to handle user interface

jmsandiegoo committed:
Implement Parser and Cmd classes to handle user commands

jmsandiegoo committed:
Merge branch 'branch-A-MoreOOP'

jmsandiegoo committed:
Implement packages in codebase

jmsandiegoo committed:
Merge branch 'branch-A-Packages'

jmsandiegoo committed:
Merge branch 'add-gradle-support'

jmsandiegoo committed:
Modify build.gradle to work with Tyrone project

jmsandiegoo committed:
Merge branch 'branch-A-Gradle'

jmsandiegoo committed:
Fix project structure and packages for testing

jmsandiegoo committed:
Rename data to task package

jmsandiegoo committed:
Add MarkCommand unit test

jmsandiegoo committed:
Add Parser unit test

jmsandiegoo committed:
Merge branch 'branch-J-Unit'

jmsandiegoo committed:
Fix minor bug in Tyrone bye output

jmsandiegoo committed:
Update autotextui script

jmsandiegoo committed:
Add find command with keyword

jmsandiegoo committed:
Merge branch 'branch-Level-9'

jmsandiegoo committed:
Add javadoc to commands package

jmsandiegoo committed:
Add javadocs to parser, storage packages

jmsandiegoo committed:
Add javadocs to storage package

jmsandiegoo committed:
Merge branch 'branch-A-JavaDoc'

jmsandiegoo committed:
Update codebase to follow coding standard

jmsandiegoo committed:
Merge branch 'branch-A-CodingStandard'

jmsandiegoo committed:
Delete duplicate test

jmsandiegoo committed:
Integrate checkstyle plugin

ys112 commented on src/main/jmsandiegoo/tyrone/common/Messages.java:
```diff
@@ -0,0 +1,66 @@
+            MESSAGE_INCORRECT_COMMAND_FORMAT,
+            "find",
+            "\"find <keyword>\""
+    );
```
I noticed the static variables in this class are using screaming snake case which violates the coding standard for a variable name. Perhaps consider changing the static variables names to camelCase or make the variable a constant if it is intended. On the side note, the bot's name and logo design is nice👍

ys112 commented on src/main/jmsandiegoo/tyrone/parser/Parser.java:
```diff
@@ -0,0 +1,208 @@
+package jmsandiegoo.tyrone.parser;
+
+import jmsandiegoo.tyrone.commands.*;
```
Could the imports be listed explicitly? (To not violate the coding standard)

ys112 reviewed (commented):
LGTM. Nice OOP structure and bot logo. I spotted some coding standard violation that could be fixed.

jmsandiegoo committed:
Add basic gui

jmsandiegoo committed:
Implement working gui and remove cli ui

jmsandiegoo committed:
Merge branch 'branch-Level-10'

xuwen-ho commented on src/main/jmsandiegoo/tyrone/Tyrone.java:
```diff
@@ -0,0 +1,130 @@
+        } catch (StorageHelperException e) {
+            throw new RuntimeException(e);
+        }
+    }
```
I like the overall OOP and organisation of the code, very easy to follow

xuwen-ho commented on src/main/resources/images/TyroneAvatar.jpeg:
Well chosen image

xuwen-ho commented on text-ui-test/EXPECTED.TXT:
```diff
@@ -1,7 +1,366 @@
+   ╚═╝      ╚═╝   ╚═╝  ╚═╝ ╚═════╝ ╚═╝  ╚═══╝╚══════╝
 
+
+Yo, what's crackin' fam! I'm Tyrone, your digital homie.
```
Love your chatbots personality

xuwen-ho reviewed (commented):
LGTM, nothing much for me to add

jmsandiegoo committed:
Add assert checks

jmsandiegoo committed:
Merge branch 'branch-A-Assertions'

jmsandiegoo committed:
General code base does not follow the code quality specified
by the se-textbook. Storage helper class has repetitive code, Task
classes has magic string literals, Parser class has long method
and more.

These coding quality / standard could lead to poor overall code
readability and quality leading to potential bugs.

Let's refactor the code base to improve the quality of the code.
Extract repetitive code to checkOrCreateFile function in StorageHelper.
Task class remove magic string literals except formatting characters.
Parser parseRawUserCommand extract checking commandWord to another
function to comply with 30 LoC. TaskList functions implement guard
clauses.

Follows the defined refactoring guidlines provided by se-textbook.
To maintain ease of maintainability.

Refer to se-textbook:
https://nus-cs2103-ay2324s2.github.io/website/se-book-adapted/chapters/codeQuality.html

jmsandiegoo committed:
Remove redundant code logic in StorageHelper class

jmsandiegoo committed:
Merge pull request #1 from jmsandiegoo/branch-A-CodeQuality

Branch a code quality

jmsandiegoo committed:
Implement undo command

User could not undo the recent commands they made in the application.

Create a new class named Undoable that follows the Memento pattern.
This class should contain the previous state and have an undo
function. Any commands eligible for undoing should implement this
undo function. Additionally, create a StateManager class to manage
history stack for the undo feature.

The code tries to follow the design pattern for such feature providing
a clear structure for managing undoable commands. Though it can be
improved in the next iteration with max stack size and not deep
copying the state for each execute and undo commands that occurs.

jmsandiegoo committed:
Merge pull request #2 from jmsandiegoo/branch-C-Undo

Implement undo command

jmsandiegoo committed:
Add user guide

jmsandiegoo committed:
Fix user guide image preview path
